### PR TITLE
Fixed lsp needs.json check

### LINKED ISSUE #603: IDE: IDE cannot find needs.json if HTML builder is used
With Sphinx-Needs 1.0.1, the IDE does not find the `needs.json`, if HTML builder is used, as it always looks for the json under
`<build_dir>/needs/needs.json`

With HTML builder, this is under `<build_dir>/html/needs.json`
> Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/pygls/protocol.py", line 331, in _handle_request
    self._execute_request(msg_id, handler, params)
  File "/usr/local/lib/python3.10/site-packages/pygls/protocol.py", line 260, in _execute_request
    method_name, method_type, msg_id, handler(params))
  File "/home/dockeruser/.local/lib/python3.10/site-packages/esbonio/lsp/__init__.py", line 216, in on_hover
    hover_value = feature.hover(context)
  File "/usr/local/lib/python3.10/site-packages/sphinx_needs/lsp/esbonio.py", line 86, in hover
    self.needs_store.load_needs(needs_json)
  File "/usr/local/lib/python3.10/site-packages/sphinx_needs/lsp/needs_store.py", line 89, in load_needs
    raise FileNotFoundError(f"JSON file not found: {json_file}")
FileNotFoundError: JSON file not found: /workspaces/sphinx-needs-basics/docs/_build/needs/needs.json

### PR DESCRIPTION
**Pull request by haiyangToAI**
fix #603, #633

### CODE REVIEW

File: sphinx_needs/lsp/esbonio.py [hunk 1]
```diff
@@ -19,10 +19,33 @@
...
+def get_needs_json(app: Sphinx) -> Optional[Path]:
+    """
+    Get the location of needs.json.
+    """
+
+    needs_json = None
+    # check if needs.json is built automatically during each sphinx-build,
+    # which requres needs_build_json = True in conf.py
+    outdir = Path(app.outdir)
+    if hasattr(app.config, "needs_build_json") and app.config.needs_build_json:
```
Comment (danieleades) [inline sphinx_needs/lsp/esbonio.py:36]: does this work instead?

```suggestion
    if getattr(app.config, "needs_build_json", False):
```

File: sphinx_needs/lsp/esbonio.py [hunk 2]
```diff
@@ -84,9 +111,13 @@ def hover(self, context: HoverContext) -> str:
         self.logger.debug(f"hover params: {context}")
 
         if isinstance(self.rst, SphinxLanguageServer) and self.rst.app:
+            # get and check needs.json path
+            needs_json = get_needs_json(self.rst.app)
+            if not needs_json or not needs_json.exists():
```
Comment (danieleades) [inline sphinx_needs/lsp/esbonio.py:116]: you can use one of de Morgan's laws here

```suggestion
            if not (needs_json and needs_json.exists()):
```

File: sphinx_needs/lsp/esbonio.py [hunk 3]
```diff
@@ -111,9 +142,13 @@ def hover(self, context: HoverContext) -> str:
     def definition(self, context: DefinitionContext) -> List[Location]:
         """Return location of definition of a need."""
         if isinstance(self.rst, SphinxLanguageServer) and self.rst.app:
+            # get and check needs.json path
+            needs_json = get_needs_json(self.rst.app)
+            if not needs_json or not needs_json.exists():
+                self.logger.warning(f"needs.json {needs_json} not exists. No Sphinx-Needs language features activated.")
```
Comment (danieleades) [inline sphinx_needs/lsp/esbonio.py:148]: deja vu. This is the same code as above, no? is it necessary to repeat the logic, or can this be reused (DRY)?

File: sphinx_needs/lsp/esbonio.py (hunk 1 above)
Comment (haiyangToAI) [reply sphinx_needs/lsp/esbonio.py:36]: I like this getattr(), it is much cleaner.

File: sphinx_needs/lsp/esbonio.py (hunk 3 above)
Comment (haiyangToAI) [reply sphinx_needs/lsp/esbonio.py:148]: unfortunately not. I know the log message it's kind of duplicate. But the line below, return type, is different for different language features in different functions.

**Review by danwos — APPROVED** [review]
Greate work. Thanks for the PR :+1:

**Pull request merged**